core/src/syn/parser/prime.rs

```diff
@@ -23,27 +25,29 @@ impl Parser<'_> {
 	/// What's are values which are more restricted in what expressions they can contain.
 	pub async fn parse_what_primary(&mut self, ctx: &mut Stk) -> ParseResult<Value> {
 		match self.peek_kind() {
+			t!("..") => Ok(self.try_parse_range(ctx, None).await?.unwrap()),
 			t!("r\"") => {
 				self.pop_peek();
-				let thing = self.parse_record_string(ctx, true).await?;
-				Ok(Value::Thing(thing))
+				let value = Value::Thing(self.parse_record_string(ctx, true).await?);
+				Ok(self.try_parse_range(ctx, Some(&value)).await?.unwrap_or(value))
```

Comment:
This is getting somewhat ugly. These functions where meant to parse a singular value so it might be better to have range be an operator and be parsed in `expression.rs` instead of having every value try to see if the next part is a range. That way it could also better participate in precedence handling.